src/components/wizard/wizard.component.ts

```diff
@@ -178,27 +183,66 @@ export class WizardComponent implements AfterViewInit, OnDestroy {
      */
     next(): void {
 
+        this.currentStep = this.getCurrentStep();
+
+        if (this.currentStep.validator) {
+
+            let validationResult = this.currentStep.validator();
+
+            const observable = typeof validationResult === 'boolean' || validationResult === null ? of(validationResult) : from(validationResult); // doesnt accept null
+
+            this.subscription = observable.pipe(first()).subscribe(
```

Comment:
I'm not really sure what all this `subscription` and `subscriptions` stuff is needed for? We have no logic in `ngOnDestroy` to tear down these subscriptions?

As per my suggestion below this could all be simplified to something like this:

```typescript
isStepValid(): Promise<boolean> {

    // get the current activer step
    this.currentStep = this.getCurrentStep();

    // if there is no validator then return the valid state
    if (!this.currentStep.validator) {
        return this.currentStep.valid;
    }

    // get the validator result
    const validatorResult = this.currentStep.validator();

    // if the function returns a promise
    return validatorResult instanceof Promise ? validatorResult : validatorResult.toPromise();
}
```

Reply:
I'm not sure, but if the wizard was destroyed while a request was pending, it might cause issues when it completes? So perhaps the subscription should be cancelled in ngOnDestroy?